src/pyhf/constraints.py

```diff
@@ -231,7 +234,17 @@ def logpdf(self, auxdata, pars):
         if self.batch_size is None:
             pois_rates = pois_rates[0]
         # pdf pars are done, now get data and compute
-        result = prob.Independent(
-            prob.Poisson(pois_rates), batch_size=self.batch_size
-        ).log_prob(self._dataprojection(auxdata))
+        result = prob.Independent(prob.Poisson(pois_rates), batch_size=self.batch_size)
+        return result
+
+    def logpdf(self, auxdata, pars):
+        tensorlib, _ = get_backend()
+        pdf = self.make_pdf(pars)
+        if pdf is None:
+            return (
+                tensorlib.zeros(self.batch_size)
+                if self.batch_size is not None
+                else tensorlib.astensor(0.0)[0]
+            )
+        result = pdf.log_prob(self._dataprojection(auxdata))
```

Comment:
Maybe just return here?

```suggestion
        return pdf.log_prob(self._dataprojection(auxdata))
```